microsoft/testsuites/acc/cvm.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features.security_profile import CvmEnabled
+from lisa.testsuite import simple_requirement
+from lisa.tools import Dmesg, Lsvmbus
+from lisa.util import LisaException
+
+
+@TestSuiteMetadata(
+    area="ACC_CVM",
+    category="functional",
+    description="""
+    This test suite ensures correct configuration and allowed devices for CVM
+    """,
+)
+class CVMSuite(TestSuite):
+    # [    0.000000] Hyper-V: Isolation Config: Group A 0x1, Group B 0xba2
+    __isolation_config_pattern = re.compile(
+        r"\[\s+\d+.\d+\]\s+Hyper-V: Isolation Config: Group A."
+        r"(?P<config_a>(0x[a-z,A-Z,0-9]+)), Group B.(?P<config_b>(0x[a-z,A-Z,0-9]+))"
+    )
+
+    @TestCaseMetadata(
+        description="""
+        This case verifies that lsvmbus only shows devices
+        that are allowed in a CVM guest
+
+        Steps:
+        1. Call lsvmbus
+        2. Iterate through list returned by lsvmbus to ensure all devices
+           listed are included in valid_class_ids
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[CvmEnabled()],
+        ),
+    )
+    def verify_lsvmbus(self, log: Logger, node: Node) -> None:
+        valid_class_ids = {
+            "ba6163d9-04a1-4d29-b605-72e2ffb1dc7f": "Synthetic SCSI Controller",
+            "f8615163-df3e-46c5-913f-f2d2f965ed0e": "Synthetic network adapter",
+            "9527e630-d0ae-497b-adce-e80ab0175caf": "[Time Synchronization]",
+            "57164f39-9115-4e78-ab55-382f3bd5422d": "[Heartbeat]",
+            "0e0b6031-5213-4934-818b-38d90ced39db": "[Operating system shutdown]",
+        }
+        lsvmbus_tool = node.tools[Lsvmbus]
+        device_list = lsvmbus_tool.get_device_channels()
+        class_id_list = [device.class_id for device in device_list]
+        assert_that(class_id_list).is_subset_of(list(valid_class_ids.keys()))
+
+    @TestCaseMetadata(
+        description="""
+        This case verifies the isolation config on guest
+
+        Steps:
+        1. Call dmesg to get output
+        2. Find isolation config in output
+        3. Check to ensure config a is 0x1
+        4. Check to ensure config b is 0xba2
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[CvmEnabled()],
+        ),
+    )
+    def verify_isolation_config(self, log: Logger, node: Node) -> None:
+        dmesg_tool = node.tools[Dmesg]
+        dmesg_output = dmesg_tool.get_output()
+        isolation_config = re.search(self.__isolation_config_pattern, dmesg_output)
+        if isolation_config is not None:
+            config_a = isolation_config.group("config_a")
```

Comment:
we'd better to add the reason for config values need to be 0x1 and 0xba2 or kernel source code. if not are these values, what is the impact.

Reply:
I agree, considering that support for other cvm types is being worked on, I expanded the testcase to
1) be easier to understand what values are being checked, and why they need to be such
2) make it easier to extend the testcase when other cvm types are added

Let me know if you think additional explanation is needed.

Reply:
isolation_config_b struct is documented in kernel code @ https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/asm-generic/mshyperv.h#n37